src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -45,12 +48,27 @@ internal sealed class TypeBuilderImpl : TypeBuilder
                 _namespace = _name[..idx];
                 _name = _name[(idx + 1)..];
             }
+
+            if (interfaces != null)
+            {
+                _interfaces = new List<Type>();
+                for (int i = 0; i < interfaces.Length; i++)
+                {
+                    // cannot contain null in the interface list
+                    ArgumentNullException.ThrowIfNull(interfaces[i], nameof(interfaces));
+                    _interfaces.Add(interfaces[i]);
```

Comment:
```suggestion
                    Type interface = interfaces[i];
                    ArgumentNullException.ThrowIfNull(interface, nameof(interfaces));
                    _interfaces.Add(interface);
```
Nit: Makes sure that the null won't sneak in if the caller modifies the array on a second thread.